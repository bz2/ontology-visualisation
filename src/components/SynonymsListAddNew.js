import React, { useState } from 'react'
import { useTranslation } from 'react-i18next'
import PropTypes from 'prop-types'
import { connect } from 'redux-zero/react'
import { InputTextarea } from 'primereact/inputtextarea'
import { Button } from 'primereact/button'
import actions from '../store/actions'
import getNode from '../utils/nodesEdgesUtils/getNode'
import synonymsCreateSynonym from '../utils/synonyms/synonymsCreateSynonym'

const SynonymsListAddNew = ({
  addNumber,
  synonymElementId,
  setStoreState,
}) => {
  const { t } = useTranslation()

  const [synonymText, setSynonymText] = useState('')
  const [showForm, setShowForm] = useState(false)

  const elementLabel = getNode(synonymElementId).label

  return (
    <div className="card">

      {!showForm
        ? (
          <div className="synonyms-synonym synonyms-add-button">
            <Button
              tooltip={t('addNewSynonym')}
              tooltipOptions={{ position: 'top' }}
              label={t('addNewSynonym')}
              id="add-synonym"
              onClick={() => setShowForm(true)}
            >
              <i className="pi pi-plus-circle" />
            </Button>
          </div>
        ) : (
          <div className="synonyms-synonym">
            <p className="bold">
              {t('addNewSynonym')}
            </p>
            <label htmlFor="synonym-textarea">{t('synonymText')}</label>
            <InputTextarea
              id="synonym-textarea"
              value={synonymText}
              type="text"
              onChange={(e) => {
                setSynonymText(e.target.value)
              }}
            />

            <p id="selected-element-label">
              {`${t('forNode')}: ${elementLabel}`}
            </p>

            <div className="synonyms-buttons">
              <Button
                tooltip={t('close')}
                tooltipOptions={{ position: 'top' }}
                className="p-button-secondary"
                label={t('close')}
                id="close-add-synonym"
                onClick={() => setShowForm(false)}
              />

              <Button
                tooltip={t('addSynonym')}
                tooltipOptions={{ position: 'top' }}
                label={t('addSynonym')}
                id="submit-synonym"
                disabled={!synonymElementId}
                onClick={() => {
                  setShowForm(false)
                  setSynonymText('')
                  synonymsCreateSynonym({
                    selectedElement: synonymElementId,
                    synonymText,
                    addNumber,
                    setStoreState,
                    t
                  })
                }}
              />
            </div>
          </div>
        )}

    </div>
  )
}

SynonymsListAddNew.propTypes = {
  synonymElementId: PropTypes.string,
  setStoreState: PropTypes.func.isRequired,
  addNumber: PropTypes.func.isRequired,
}

SynonymsListAddNew.defaultProps = {
  synonymElementId: undefined,
}

const mapToProps = ({
  synonyms,
  synonymElementId,
  nodesNotes,
  edgesNotes,
  user
}) => ({
  synonyms,
  synonymElementId,
  nodesNotes,
  edgesNotes,
  user
})

export default connect(
  mapToProps,
  actions
)(SynonymsListAddNew)
