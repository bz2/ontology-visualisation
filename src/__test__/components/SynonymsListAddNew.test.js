import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import SynonymsListAddNew from '../../components/SynonymsListAddNew'

const setup = () => {
  const props = {
    synonymElementId: '12',
    updateStoreValue: jest.fn(),
  }

  const component = shallow(<SynonymsListAddNew {...props} />)

  return {
    component,
    props
  }
}

describe('SynonymsListAddNew', () => {
  it('should match snapshot', () => {
    const {
      component
    } = setup()

    expect(toJson(component)).toMatchSnapshot()
  })
})
