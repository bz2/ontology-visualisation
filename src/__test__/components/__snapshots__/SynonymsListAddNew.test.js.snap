// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SynonymsListAddNew should match snapshot 1`] = `
<Connect
  actions={
    Object {
      "updateStoreValue": [Function],
    }
  }
  mapToProps={[Function]}
  synonymElementId="12"
  updateStoreValue={[MockFunction]}
>
  <Component />
</Connect>
`;
